docs/analytical_engine/tutorial_dev_algo_cpp_pie.md

```diff
@@ -1,2 +1,148 @@
 # Tutorial: Develop your Algorithm in C++ with PIE Model
 
+In this tutorial, we will demonstrate how to develop an algorithm using the PIE model in C++ programming language.
+
+## Prerequisites:
+- Familiarity with the [PIE programming model](https://graphscope.io/docs/latest/analytical_engine/programming_model_pie.html) 
+- Install GraphScope by using the command `pip3 install graphscope`
+
+Then, We will guide you through developing a simple PIE algorithm that computes the degree for each vertex in the graph. The full API can be found in [Analytical Engine API Doc](https://graphscope.io/docs/latest/reference/analytical_engine_index.html) and [libgrape-lite API](https://alibaba.github.io/libgrape-lite).
+
+## Step 1: Define the context class
+
+First, we create a context class that inherits from `grape::VertexDataContext`. This class will store and manage algorithm-specific data and parameters.
+
+```cpp
+template <typename FRAG_T>
+class MyAppContext : public grape::VertexDataContext<FRAG_T, uint64_t> {
+  using oid_t = typename FRAG_T::oid_t;
+  using vid_t = typename FRAG_T::vid_t;
+  using vertex_t = typename FRAG_T::vertex_t;
+
+ public:
+  explicit MyAppContext(const FRAG_T& fragment)
+      : grape::VertexDataContext<FRAG_T, uint64_t>(fragment, true),
+        result(this->data()) {}
+
+  void Init(grape::ParallelMessageManager& messages, int param1) {
+    this->step = 0;
+    this->param1 = param1;
+    result.SetValue(0);
+  }
+
+  int step = 0;
+  int param1 = 0;
+  typename FRAG_T::template vertex_array_t<uint64_t>& result;
+};
+```
+
+As shown in the code. Respectively, The `MyAppContext` class defines two member variables called `step` and `param1` to store the current superstep and algorithm-specific parameter. And also defines a member variable named `result` with `uint64_t` type to store the the degree for each vertex in the fragment. The `Init` method is used to initialize the context of the computation. In current example, we initialize the `step` and `param1` variables to zero and the algorithm-specific parameter. We also set the result to zero for each vertex.
```

Comment:
As shown in the code, the `MyAppContext` class defines two member variables called `step` and `param1` to store the current superstep and algorithm-specific parameter, respectively. And we also define a member variable named `result` with `uint64_t` type to store the the degree for each vertex in the fragment.